routes/router.go

```diff
@@ -12,5 +13,6 @@ RegisterRoutes is a place to register rotes
 func RegisterRoutes(router *gin.Engine) {
 
 	router.POST("/login", login.Controller)
-
+	router.POST("/curriculums", curriculum.Controller)
```

Comment:
It is not necessary to open this API, instead, it should log in itself when it needs.

Reply:
In my opinion, it should look like
1. NportalConnector
    - isLogin(): bool
    - login(): void
    
2. CourseConnector
    - isLogin()
    - login()
    - getCourseSemesters(String sid): Semester[]
    - getStudentCourse(String sid, String year, String semester): StudentCourse
    - getClassmate(String courseNo): String[]
    - getCourseType(String courseNo): String
    ......

Reply:
```
if (!NportalConnector.isLogin()) {
    NportalConnector.login(account, password);
}
if (!CourseConnector.isLogin()) {
    CourseConnector.loginCourse();
}
result = CourseConnector.getStudentCourse(...);
```